packages/commonwealth/server/routes/bulkBalances.ts

```diff
@@ -0,0 +1,131 @@
+import TokenBalanceCache from 'token-balance-cache/src/index';
+import { AddressInstance } from 'server/models/address';
+import { DB } from '../database';
+import { AppError, ServerError } from '../util/errors';
+import { success, TypedRequestBody, TypedResponse } from '../types';
+
+enum BulkBalancesErrors {
+  NoProfileId = 'No profileId provided',
+  NoProfile = "Profile doesn't exist",
+  NoRegisteredAddresses = 'No wallet addresses registered to this profile',
+  TokenContractNotDeployed = 'Token contract not deployed on specified chain node', // Currently not used, see point 3
+}
+
+// ----------- OUTSTANDING QUESTIONS -----------
+// 1. Max across addresses or sum across addresses?
+// 2. Handling token addresses not deployed on a Chain Node? Currently just ignoring them and still returning a success
+// 3. How is this route being authenticated, if at all?
```

Comment:
For #3, we could require an authorization basic header with the `DISCORD_BOT_TOKEN` discord api key as the token.

Reply:
Makes sense to me, @jnaviask any objections?

Reply:
Sounds good to me.